src/main/java/spoon/support/compiler/jdt/JDTTreeBuilder.java

```diff
@@ -1175,9 +1175,26 @@ public boolean visit(IntLiteral intLiteral, BlockScope scope) {
 
 	@Override
 	public boolean visit(LabeledStatement labeledStatement, BlockScope scope) {
-		context.label.push(labeledStatement);
+		//context.enter(factory.Core().createBlock(), block);
+		ContextBuilder.LabeledStatementHolder outerLabelHolder = context.label.peek();
+		if (outerLabelHolder != null && outerLabelHolder.wasAssignedToAStatement == false) {
+			/*
+			 * It is the case of nested label like:
+			 * label1: label2: while(x) {}
+			 * We have to add implicit block
+			 * label1: {label2: while(x) {}}
+			 */
+			how to create implicit CtBlock? Does it belong here or somewhere else?
```

Comment:
how to create implicit CtBlock? Does it belong here or somewhere else?
@tdurieux Could you implement it or give me a tip?